grakn-engine/src/main/java/ai/grakn/engine/controller/api/AttributeTypeController.java

```diff
@@ -0,0 +1,144 @@
+        String attributeTypeDataTypeRaw = requestBody.at("attributeType").at("type").asString();
+        AttributeType.DataType<?> attributeTypeDataType = fromString(attributeTypeDataTypeRaw);
+        String keyspace = mandatoryQueryParameter(request, KEYSPACE);
+        LOG.info("postAttributeType - attempting to add new attributeType " + attributeTypeLabel + " of type " + attributeTypeDataTypeRaw);
+        try (GraknTx tx = factory.tx(Keyspace.of(keyspace), GraknTxType.WRITE)) {
+            AttributeType attributeType = tx.putAttributeType(attributeTypeLabel, attributeTypeDataType);
+            tx.commit();
+            String jsonConceptId = attributeType.getId().getValue();
+            String jsonAttributeTypeLabel = attributeType.getLabel().getValue();
+            String jsonAttributeTypeDataType = toString(attributeType.getDataType());
+            LOG.info("postAttributeType - attribute type " + jsonAttributeTypeLabel +
+                " of type " + jsonAttributeTypeDataType + " with id " + jsonConceptId + " added successfully. request processed.");
+            response.status(HttpStatus.SC_OK);
+            Json responseBody = attributeTypeJson(jsonConceptId, jsonAttributeTypeLabel, jsonAttributeTypeDataType);
+            return responseBody;
+        }
+    }
+
+    private Json getAttributeType(Request request, Response response) {
+        LOG.info("getAttributeType - request received.");
+        String attributeTypeLabel = mandatoryPathParameter(request, "attributeTypeLabel");
+        String keyspace = mandatoryQueryParameter(request, KEYSPACE);
+        LOG.info("getAttributeType - attempting to find attributeType " + attributeTypeLabel + " in keyspace " + keyspace);
+        try (GraknTx tx = factory.tx(Keyspace.of(keyspace), GraknTxType.READ)) {
+            Optional<AttributeType> attributeType = Optional.ofNullable(tx.getAttributeType(attributeTypeLabel));
+            if (attributeType.isPresent()) {
+                String jsonConceptId = attributeType.get().getId().getValue();
+                String jsonAttributeTypeLabel = attributeType.get().getLabel().getValue();
+                String jsonAttributeTypeDataType = toString(attributeType.get().getDataType());
+                response.status(HttpStatus.SC_OK);
+                Json responseBody = attributeTypeJson(jsonConceptId, jsonAttributeTypeLabel, jsonAttributeTypeDataType);
+                LOG.info("getAttributeType - attributeType found - " + jsonConceptId + ", " +
+                    jsonAttributeTypeLabel + ", " + jsonAttributeTypeDataType + ". request processed.");
+                return responseBody;
+            } else {
+                response.status(HttpStatus.SC_BAD_REQUEST);
+                LOG.info("getAttributeType - attributeType NOT found. request processed.");
+                return Json.nil();
+            }
+        }
+    }
+
+    private AttributeType.DataType<?> fromString(String dataType) {
```

Comment:
Yup, confirmed that we will need this method. Otherwise it will expect `java.lang.String`, `java.lang.Boolean` instead of `string`, `boolean`, etc. In other words, exposing implementation details and is inappropriate for a public API.

Reply:
Ah okay that makes sense